src/beacon/app/data_transmission_endpoint.py

```
@@ -1,499 +1,732 @@
+        if time_range == TimeRangeEnum.CUSTOM and start_date and end_date:
+            try:
+                start_dt = datetime.strptime(start_date, "%Y-%m-%d")
+                end_dt = datetime.strptime(end_date, "%Y-%m-%d")
+                
+                # Localize to specified timezone
+                start_dt = timezone.localize(start_dt)
+                end_dt = timezone.localize(end_dt.replace(hour=23, minute=59, second=59))
+                
+                return start_dt, end_dt
+            except ValueError:
+                raise HTTPException(status_code=400, detail="Invalid date format. Use YYYY-MM-DD")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)